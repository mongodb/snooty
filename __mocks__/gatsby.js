const gatsby = jest.requireActual('gatsby');

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  withPrefix: jest.fn().mockImplementation(str => str),
  useStaticQuery: jest.fn(),
};
