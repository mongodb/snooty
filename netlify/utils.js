import fs from 'fs';
import path from 'path';

/**
 * Parses build hook data from the expected temporary location
 * @returns {object | undefined}
 */
const parseBuildHookData = () => {
  // This file does not currently exist, but should be created on Netlify builds as part of `npm run build:netlify`.
  // The INCOMING_HOOK_BODY env var is not automatically passed along to functions, so we use a txt file to save it
  const relativeFilePath = '../build-hook.txt';
  const buildHookDataString = fs.readFileSync(path.resolve(__dirname, relativeFilePath), 'utf-8');
  if (!buildHookDataString) {
    console.log('No build hook data found.');
    return;
  }
  return JSON.parse(buildHookDataString);
};

export const constructResPayload = (event) => {
  const buildHookData = parseBuildHookData();
  const parsedEventBody = JSON.parse(event.body);
  // This is Netlify's default post-deployment payload. We include it with our custom data in case
  // we want to process any information
  const netlifyPayload = parsedEventBody.payload;
  return {
    netlifyPayload,
    ...buildHookData,
  };
};
