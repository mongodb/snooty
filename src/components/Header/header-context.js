import React, { createContext, useEffect, useState } from 'react';
import { theme } from '../../theme/docsTheme';

const HeaderContext = createContext({
  bannerContent: null,
  setBannerContent: () => {},
  totalHeaderHeight: theme.header.navbarHeight,
});

const HeaderContextProvider = ({ children }) => {
  const [bannerContent, setBannerContent] = useState(null);
  const [totalHeaderHeight, setTotalHeaderHeight] = useState(theme.header.navbarHeight);

  useEffect(() => {
    const totalHeight = `calc(
      ${bannerContent != null ? `${theme.header.bannerHeight} +` : ''}
      ${theme.header.navbarHeight}
    )`;
    setTotalHeaderHeight(totalHeight);
  }, [bannerContent, setTotalHeaderHeight]);

  return (
    <HeaderContext.Provider value={{ bannerContent, setBannerContent, totalHeaderHeight }}>
      {children}
    </HeaderContext.Provider>
  );
};

export { HeaderContext, HeaderContextProvider };
