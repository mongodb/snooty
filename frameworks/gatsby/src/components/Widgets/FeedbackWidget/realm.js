import React from 'react';
import * as Realm from 'realm-web';
import { ObjectID } from 'bson';
import { isBrowser } from '../../../utils/is-browser';
import { removeAllRealmUsersFromLocalStorage } from '../../../utils/realm-user-management';

const APP_ID = 'feedbackwidgetv3-dgcsv';
export const app = isBrowser ? Realm.App.getApp(APP_ID) : { auth: {} };

/**
 * Deletes localStorage data for all users
 */
function deleteLocalStorageData() {
  const { allUsers } = app;
  removeAllRealmUsersFromLocalStorage(allUsers);
}

// User Authentication & Management
export async function loginAnonymous() {
  if (!app.currentUser) {
    const user = await app.logIn(Realm.Credentials.anonymous());
    return user;
  }
  return app.currentUser;
}

export async function logout() {
  if (app.auth.isLoggedIn) {
    await app.auth.logoutUserWithId(app.id);
  } else {
    console.warn('No logged in user.');
  }
}

export const useRealmUser = () => {
  const [user, setUser] = React.useState(app.currentUser);

  async function reassignCurrentUser() {
    const oldUser = app.currentUser;
    try {
      await app.removeUser(oldUser);
    } catch (e) {
      console.error(e);
    }

    // Clean up invalid data from local storage to avoid bubbling up local storage sizes for broken user credentials
    // This should be safe since only old users' data would be deleted, and we make a new user right after
    deleteLocalStorageData();

    const newUser = await app.logIn(Realm.Credentials.anonymous());
    setUser(newUser);
    return newUser;
  }

  // Initial user login
  React.useEffect(() => {
    loginAnonymous()
      .then(setUser)
      .catch((e) => {
        console.error(e);
      });
  }, []);

  return { user, reassignCurrentUser };
};

// Feedback Widget Functions
export async function upsertFeedback({ page, user, attachment, ...rest }) {
  const updateOneRes = await app.currentUser.callFunction('feedback_upsert', { page, user, attachment, ...rest });
  const objectId = new ObjectID(updateOneRes.upsertedId);
  return objectId.toString();
}
