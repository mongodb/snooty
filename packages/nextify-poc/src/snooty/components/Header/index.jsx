import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css, cx } from '@leafygreen-ui/emotion';
import { UnifiedNav } from '@mdb/consistent-nav';
import SiteBanner from '../Banner/SiteBanner';
import { theme } from '../../theme/docsTheme';
import { getAvailableLanguages, getCurrLocale, onSelectLocale } from '../../utils/locale';
import { isOfflineDocsBuild } from '../../utils/is-offline-docs-build';
import { HeaderContext } from './header-context';

const StyledHeaderContainer = styled.header(
  (props) => `
  grid-area: header;
  top: 0;
  margin-top: ${props.hasBanner ? theme.header.bannerHeight : '0px'};
  z-index: ${theme.zIndexes.header};
  `
);

const offlineClass = css`
  .header-desktop-buttons {
    display: none;
  }

  button[aria-label='Open Links'] {
    display: none;
  }
`;

const Header = ({ eol, template }) => {
  const unifiedNavProperty = 'DOCS';

  const enabledLocales = getAvailableLanguages().map((language) => language.localeCode);
  const { bannerContent } = useContext(HeaderContext);

  return (
    <>
      <SiteBanner />
      <StyledHeaderContainer template={template} hasBanner={!!bannerContent}>
        <>
          {/* Two navs used intentionally: one for light mode, one for dark mode */}
          {!eol && (
            <>
              <UnifiedNav
                fullWidth="true"
                hideSearch="true"
                position="relative"
                property={{ name: unifiedNavProperty }}
                showLanguageSelector={true}
                onSelectLocale={onSelectLocale}
                locale={getCurrLocale()}
                enabledLocales={enabledLocales}
                darkMode={false}
                className={cx('nav-light', isOfflineDocsBuild ? offlineClass : '')}
              />
              <UnifiedNav
                fullWidth="true"
                hideSearch="true"
                position="relative"
                property={{ name: unifiedNavProperty }}
                showLanguageSelector={true}
                onSelectLocale={onSelectLocale}
                locale={getCurrLocale()}
                enabledLocales={enabledLocales}
                darkMode={true}
                className={cx('nav-dark', isOfflineDocsBuild ? offlineClass : '')}
              />
            </>
          )}
        </>
      </StyledHeaderContainer>
    </>
  );
};

Header.propTypes = {
  eol: PropTypes.bool.isRequired,
  template: PropTypes.string.isRequired,
};

export default Header;
