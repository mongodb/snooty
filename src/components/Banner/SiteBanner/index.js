import React, { useContext, useEffect } from 'react';
import styled from '@emotion/styled';
import { palette } from '@leafygreen-ui/palette';
import { css } from '@leafygreen-ui/emotion';
import { HeaderContext } from '../../Header/header-context';
import { SNOOTY_REALM_APP_ID } from '../../../build-constants';
import { useSiteMetadata } from '../../../hooks/use-site-metadata';
import { theme } from '../../../theme/docsTheme';
import { isBrowser } from '../../../utils/is-browser';
import { normalizePath } from '../../../utils/normalize-path';
import { fetchBanner } from '../../../utils/realm';
import BrandingShape from './BrandingShape';

const getBannerSource = (src) => {
  if (src == null || src === '') return null;
  const srcUrl = `${SNOOTY_REALM_APP_ID}.mongodbstitch.com/${src}`;
  return `https://${normalizePath(srcUrl)}`;
};

const StyledBannerContainer = styled.a`
  display: block;
  height: ${theme.header.bannerHeight};
  width: 100%;
  position: absolute;
  z-index: ${theme.zIndexes.header};
  color: white;
  text-decoration: none;
`;

const StyledBannerContent = styled.div(
  (props) => `
    background-image: url(${getBannerSource(props.imgPath)});
    background-position: center;
    background-size: cover;
    background-color: ${props.bgColor};
    height: 100%;
    display: flex;
    justify-content: center;
    gap: 20px;
    padding: 0 11px;

    @media ${theme.screenSize.upToMedium} {
      background-image: url(${getBannerSource(props.tabletImgPath)});
    }

    @media ${theme.screenSize.upToSmall} {
      background-image: url(${getBannerSource(props.mobileImgPath)});
    }
  `
);

const bannerTextStyle = css`
  align-self: center;
`;

const pillContainer = css`
  display: grid;
  justify-items: center;
  align-items: center;
`;

// Forces components to be in the same cell to create an overlap
const gridCell = css`
  grid-row: 1;
  grid-column: 1;
`;

const brandingContainer = css`
  ${gridCell}
  height: 40px;
`;

const pillStyle = css`
  ${gridCell}
  color: ${palette.green.dark3};
  font-weight: 600;
  line-height: 16px;
  font-size: 12px;
  background-color: ${palette.green.base};
  border: 1px solid ${palette.green.dark2};
  border-radius: 6px;
  height: 22px;
  padding: 3px 8px;
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const SiteBanner = () => {
  const { bannerContent, setBannerContent } = useContext(HeaderContext);
  const { snootyEnv } = useSiteMetadata();

  useEffect(() => {
    const fetchBannerContent = async () => {
      try {
        setBannerContent(await fetchBanner(snootyEnv));
        // TODO-5132: Remove and test with App Services when done
        const bannerContentMock = {
          text: 'Join us at AWS re:Invent 2024! Learn how to use MongoDB for AI use cases.',
          altText: 'Hello World!',
          pillText: 'LEARN MORE',
          bgColor: palette.green.dark3,
          url: 'https://www.mongodb.com/docs/',
        };
        setBannerContent(bannerContentMock);
      } catch (err) {
        console.error(err);
      }
    };
    if (isBrowser) {
      fetchBannerContent();
    }
  }, [setBannerContent, snootyEnv]);

  if (bannerContent == null) {
    return null;
  }

  return (
    <StyledBannerContainer href={bannerContent.url} title={bannerContent.altText}>
      <StyledBannerContent
        imgPath={bannerContent.imgPath}
        tabletImgPath={bannerContent.tabletImgPath ?? bannerContent.mobileImgPath}
        mobileImgPath={bannerContent.mobileImgPath}
        bgColor={bannerContent.bgColor}
      >
        {bannerContent.text && (
          <>
            <span className={bannerTextStyle}>{bannerContent.text}</span>
            <div className={pillContainer}>
              <div className={brandingContainer}>
                <BrandingShape size={'desktop'} />
              </div>
              {bannerContent.pillText && <span className={pillStyle}>{bannerContent.pillText}</span>}
            </div>
          </>
        )}
      </StyledBannerContent>
    </StyledBannerContainer>
  );
};

export default SiteBanner;
