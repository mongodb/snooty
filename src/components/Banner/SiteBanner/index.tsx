import React, { useContext, useEffect } from 'react';
import { palette } from '@leafygreen-ui/palette';
import { css, cx } from '@leafygreen-ui/emotion';
import { HeaderContext } from '../../Header/header-context';
import { SNOOTY_REALM_APP_ID } from '../../../build-constants';
import { useSiteMetadata } from '../../../hooks/use-site-metadata';
import { theme } from '../../../theme/docsTheme';
import { isBrowser } from '../../../utils/is-browser';
import { normalizePath } from '../../../utils/normalize-path';
import { fetchBanner } from '../../../utils/realm';
import { SiteBannerContent } from './types';
import BrandingShape from './BrandingShape';

const getBannerSource = (src?: string) => {
  if (src == null || src === '') return null;
  const srcUrl = `${SNOOTY_REALM_APP_ID}.mongodbstitch.com/${src}`;
  return `https://${normalizePath(srcUrl)}`;
};

const bannerContainerStyle = css`
  display: block;
  height: ${theme.header.bannerHeight};
  width: 100%;
  position: absolute;
  z-index: ${theme.zIndexes.header};
  color: white;
  text-decoration: none;
`;

const bannerContentStyle = (bannerContent: Partial<SiteBannerContent>) => css`
  background-image: url(${getBannerSource(bannerContent.imgPath)});
  background-position: center;
  background-size: cover;
  ${bannerContent.bgColor && `background-color: ${bannerContent.bgColor};`}
  height: 100%;
  display: flex;
  justify-content: center;
  gap: 20px;
  padding: 0 11px;
  font-size: ${theme.fontSize.small};
  line-height: 20px;

  @media ${theme.screenSize.upToMedium} {
    background-image: url(${getBannerSource(bannerContent.tabletImgPath)});
    justify-content: space-between;
  }

  @media ${theme.screenSize.upToSmall} {
    background-image: url(${getBannerSource(bannerContent.mobileImgPath)});
    font-size: ${theme.fontSize.xsmall};
  }
`;

const bannerTextStyle = css`
  align-self: center;
  max-height: 40px;
`;

const pillContainer = css`
  display: grid;
  justify-items: center;
  align-items: center;
`;

// Forces components to be in the same cell to create an overlap
const gridCell = css`
  grid-row: 1;
  grid-column: 1;
`;

const brandingContainer = css`
  ${gridCell}
  height: 40px;
`;

const pillStyle = css`
  ${gridCell}
  color: ${palette.green.dark3};
  font-weight: 600;
  line-height: 16px;
  font-size: ${theme.fontSize.tiny};
  background-color: ${palette.green.base};
  border: 1px solid ${palette.green.dark2};
  border-radius: 6px;
  height: 22px;
  padding: 3px 8px;
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const SiteBanner = () => {
  const { bannerContent, setBannerContent } = useContext(HeaderContext);
  const { snootyEnv } = useSiteMetadata();

  useEffect(() => {
    const fetchBannerContent = async () => {
      try {
        const res: SiteBannerContent | null = await fetchBanner(snootyEnv);
        // Guard against missing banner content
        if (res && res.url && (res.imgPath || res.text)) {
          setBannerContent(res);
        }
      } catch (err) {
        console.error(err);
      }
    };
    if (isBrowser) {
      fetchBannerContent();
    }
  }, [setBannerContent, snootyEnv]);

  if (!bannerContent) {
    return null;
  }

  // Ensure Smartling doesn't translate the banner or rewrite anything
  const smartlingClassNames = 'sl_opaque notranslate';
  // Backup class name in case Smartling needs to target the whole element
  const bannerClassName = 'site-banner';

  return (
    <a
      className={cx(bannerClassName, smartlingClassNames, bannerContainerStyle)}
      href={bannerContent.url}
      title={bannerContent.altText}
    >
      <div
        className={bannerContentStyle({
          imgPath: bannerContent.imgPath,
          tabletImgPath: bannerContent.tabletImgPath ?? bannerContent.mobileImgPath,
          mobileImgPath: bannerContent.mobileImgPath,
          bgColor: bannerContent.bgColor,
        })}
      >
        {bannerContent.text && (
          <>
            <span className={cx(smartlingClassNames, bannerTextStyle)}>{bannerContent.text}</span>
            <div className={pillContainer}>
              <div className={brandingContainer}>
                <BrandingShape />
              </div>
              {bannerContent.pillText && (
                <span className={cx(smartlingClassNames, pillStyle)}>{bannerContent.pillText}</span>
              )}
            </div>
          </>
        )}
      </div>
    </a>
  );
};

export default SiteBanner;
