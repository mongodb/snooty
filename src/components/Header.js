import React from 'react';
import styled from '@emotion/styled';
import SiteBanner from './SiteBanner';
import Navbar from './Navbar';

const StyledHeaderContainer = styled.header`
  grid-area: header;
  z-index: 10;
`;

const Header = () => {
  return (
    <StyledHeaderContainer>
      <SiteBanner />
      <Navbar />
    </StyledHeaderContainer>
  );
};

export default Header;
