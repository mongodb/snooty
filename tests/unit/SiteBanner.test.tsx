import React from 'react';
import { render } from '@testing-library/react';
import * as Gatsby from 'gatsby';
import { palette } from '@leafygreen-ui/palette';
import * as RealmUtil from '../../src/utils/realm';
import SiteBanner from '../../src/components/Banner/SiteBanner';
import { HeaderContext } from '../../src/components/Header/header-context';
import { tick } from '../utils';

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
const mockSnootyEnv = (snootyEnv: string) => {
  useStaticQuery.mockImplementation(() => ({
    site: {
      siteMetadata: {
        snootyEnv,
      },
    },
  }));
};

const mockBannerContent = {
  isEnabled: true,
  altText: 'Test',
  imgPath: '/banners/test.png',
  tabletImgPath: '/banners/test-tablet.png',
  mobileImgPath: '/banners/test-mobile.png',
  url: 'https://mongodb.com',
};

describe('Banner component', () => {
  beforeAll(() => {
    mockSnootyEnv('development');
  });

  it('renders without a banner image', () => {
    // bannerContent state should remain null
    const wrapper = render(<SiteBanner />);
    expect(wrapper.queryByAltText(mockBannerContent.altText)).toBeNull();
  });

  it('renders with a banner image', async () => {
    jest.useFakeTimers();
    jest.spyOn(RealmUtil, 'fetchBanner').mockResolvedValueOnce(() => mockBannerContent);
    const setBannerContent = jest.fn();
    const wrapper = render(
      <HeaderContext.Provider
        value={{ bannerContent: mockBannerContent, setBannerContent: setBannerContent, totalHeaderHeight: '' }}
      >
        <SiteBanner />
      </HeaderContext.Provider>
    );
    await tick();
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('renders with custom text', async () => {
    jest.useFakeTimers();
    const bannerContent = {
      isEnabled: true,
      altText: mockBannerContent.altText,
      bgColor: palette.green.dark3,
      text: 'This is custom banner text',
      pillText: 'DOP',
      url: mockBannerContent.url,
    };
    jest.spyOn(RealmUtil, 'fetchBanner').mockResolvedValueOnce(() => bannerContent);
    const setBannerContent = jest.fn();
    const wrapper = render(
      <HeaderContext.Provider value={{ bannerContent, setBannerContent: setBannerContent, totalHeaderHeight: '' }}>
        <SiteBanner />
      </HeaderContext.Provider>
    );
    await tick();
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
