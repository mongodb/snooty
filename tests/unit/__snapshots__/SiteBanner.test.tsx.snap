// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders with a banner image 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: block;
  height: 40px;
  width: 100%;
  position: absolute;
  z-index: 1000;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 {
  background-image: url(https://snooty-koueq.mongodbstitch.com/banners/test.png);
  -webkit-background-position: center;
  background-position: center;
  -webkit-background-size: cover;
  background-size: cover;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 20px;
  padding: 0 11px;
  font-size: 13px;
  line-height: 20px;
}

@media only screen and (max-width: 767px) {
  .emotion-1 {
    background-image: url(https://snooty-koueq.mongodbstitch.com/banners/test-tablet.png);
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

@media only screen and (max-width: 480px) {
  .emotion-1 {
    background-image: url(https://snooty-koueq.mongodbstitch.com/banners/test-mobile.png);
    font-size: 11px;
  }
}

<a
    class="site-banner sl_opaque notranslate emotion-0"
    href="https://mongodb.com"
    title="Test"
  >
    <div
      class="emotion-1"
    />
  </a>
</DocumentFragment>
`;

exports[`Banner component renders with custom text 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: block;
  height: 40px;
  width: 100%;
  position: absolute;
  z-index: 1000;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 {
  background-image: url();
  -webkit-background-position: center;
  background-position: center;
  -webkit-background-size: cover;
  background-size: cover;
  background-color: #023430;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 20px;
  padding: 0 11px;
  font-size: 13px;
  line-height: 20px;
}

@media only screen and (max-width: 767px) {
  .emotion-1 {
    background-image: url();
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    justify-content: space-between;
  }
}

@media only screen and (max-width: 480px) {
  .emotion-1 {
    background-image: url();
    font-size: 11px;
  }
}

.emotion-2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  max-height: 40px;
}

.emotion-3 {
  display: grid;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  grid-row: 1;
  grid-column: 1;
  height: 40px;
}

.emotion-5 {
  grid-row: 1;
  grid-column: 1;
  color: #023430;
  font-weight: 600;
  line-height: 16px;
  font-size: 12px;
  background-color: #00ED64;
  border: 1px solid #00684A;
  border-radius: 6px;
  height: 22px;
  padding: 3px 8px;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<a
    class="site-banner sl_opaque notranslate emotion-0"
    href="https://mongodb.com"
    title="Test"
  >
    <div
      class="emotion-1"
    >
      <span
        class="sl_opaque notranslate emotion-2"
      >
        This is custom banner text
      </span>
      <div
        class="emotion-3"
      >
        <div
          class="emotion-4"
        >
          <svg
            fill="none"
            height="40"
            viewBox="0 0 109 40"
            width="109"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M55.9997 98.9928L55.9997 99.0001L55.9998 151C55.9998 165.076 44.5756 176.5 30.4998 176.5C16.4239 176.5 4.99975 165.076 4.99975 151C4.99975 136.372 -6.8721 124.5 -21.5002 124.5L-73.5002 124.5C-87.5761 124.5 -99.0002 113.076 -99.0003 99.0001C-99.0003 84.9242 -87.5761 73.5001 -73.5003 73.5001L-47.2923 73.5001C-18.3108 73.5001 4.99974 49.9789 4.99974 21.208L4.99974 -4.99991C4.99974 -19.0758 16.4239 -30.4999 30.4997 -30.4999C44.5756 -30.4999 55.9997 -19.0758 55.9997 -4.99992C55.9997 9.62821 67.8716 21.5001 82.4997 21.5001C96.5756 21.5001 108 32.9242 108 47.0001C108 61.0759 96.5756 72.5001 82.4997 72.5001C67.8651 72.5001 56.2116 84.377 55.9997 98.9928Z"
              stroke="#00ED64"
            />
          </svg>
        </div>
        <span
          class="sl_opaque notranslate emotion-5"
        >
          DOP
        </span>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
